### ISSUE TITLE
Target branch is not autodetected

### ISSUE THREAD

**Issue by jooola** [google-github-actions/release-please-action#913]
### TL;DR

The target branch is not autodetected as described by the action documentation.

### Expected behavior

The target branch should be autodetected by either release-please or the release-please-action.

### Observed behavior

We run release-please on a version branch (`release-1.x`), and the target branch in the logs is still `main`.

### Action YAML

```yaml
name: Release-please

on:
  push:
    branches:
      - main
      - release-1.x

jobs:
  release-please:
    # Do not run on forks.
    if: github.repository == 'hetznercloud/hcloud-go'

    runs-on: ubuntu-latest
    steps:
      - uses: google-github-actions/release-please-action@v4
        with:
          config-file: .github/release-please-config.json
          manifest-file: .github/release-please-manifest.json
```

### Log output

```text
Run google-github-actions/release-please-action@v4
  with:
    config-file: .github/release-please-config.json
    manifest-file: .github/release-please-manifest.json
    token: ***
    repo-url: hetznercloud/hcloud-go
    github-api-url: https://api.github.com
    github-graphql-url: https://api.github.com/graphql
    fork: false
    include-component-in-tag: false
    skip-github-release: false
    skip-github-pull-request: false
Running release-please version: 16.5.0
❯ Fetching .github/release-please-config.json from branch main
❯ Fetching .github/release-please-manifest.json from branch main
✔ Building releases
✔ Building strategies by path
❯ .: go
❯ Fetching .github/release-please-config.json from branch main
❯ Fetching .github/release-please-manifest.json from branch main
✔ Building pull requests
✔ Building strategies by path
❯ .: go
✔ Collecting release commit SHAs
❯ release search depth: 400
❯ Fetching releases with cursor undefined
❯ Found release for path ., v2.5.1
❯ release for path: ., version: 2.5.1, sha: 607142011735c29af438d62666dc63031de9c310
✔ Collecting commits since all latest releases
❯ commit search depth: 500
❯ Set(1) { '607142011735c29af438d62666dc63031de9c310' }
❯ Fetching merge commits on branch main with cursor: undefined
❯ Backfilling file list for commit: 851a65e23dbb1d49fc3aa95adae184e47bde32c9
❯ Found 2 files
✔ Splitting 1 commits by path
✔ Building candidate release pull request for path: .
❯ type: go
❯ targetBranch: main
❯ commits: 1
✔ Considering: 1 commits
❯ component: 
❯ pull request title pattern: undefined
✔ No user facing commits found since 607142011735c29af438d62666dc63031de9c310 - skipping
```

### Additional information

Some links:
- https://github.com/hetznercloud/hcloud-go/actions/runs/7262147825/job/19784717793
- https://github.com/hetznercloud/hcloud-go/blob/release-1.x/.github/workflows/release-please.yml
- https://github.com/hetznercloud/hcloud-go/blob/release-1.x/.github/release-please-config.json
- https://github.com/hetznercloud/hcloud-go/blob/release-1.x/.github/release-please-manifest.json
